app/src/main/java/org/mozilla/tv/firefox/pocket/PocketVideoRepo.kt

```diff
@@ -39,102 +29,40 @@ open class PocketVideoRepo(
         object Inactive : FeedState()
     }
 
-    private val _feedState = MutableLiveData<FeedState>().apply {
-        // setValue must be called from the main thread, and this repo is instantiated
-        // off of the main thread in Espresso tests, so postValue is used instead
-        postValue(buildConfigDerivables.initialPocketRepoState)
-    }
-    open val feedState: LiveData<FeedState> = _feedState
+    private val _feedState = BehaviorSubject.createDefault(initialState)
```

Comment:
one nice way to avoid the `_`-prefixed variables is:

```kotlin
open val feedState: Observable<PocketVideoRepo.FeedState> = BehaviorSubject.createDefault(initialState)
```

Then, when you need to push to the `feedState`, you can cast to `as Subject` below.

Reply:
My only concern with that approach is that downstream objects would be able to cast it to a subject as well.

Reply:
> My only concern with that approach is that downstream objects would be able to cast it to a subject as well.

@Baron-Severin what's the downside of that?

Reply:
Subjects can be observed, but they can also be pushed to (they implement both Observable and Observer interfaces).  This means they're a lot like mutable properties: sometimes necessary, but always dangerous, and should be kept as narrowly scoped as possible.